inc/Plugin.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Bluehost\Maestro;
+
+/**
+ * Class for managing plugins
+ */
+class Plugin {
+	/**
+	 * Plugin's slug
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	public $slug;
+
+	/**
+	 * Plugin's name
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	public $name;
+
+
+	/**
+	 * Plugin's version
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	public $version;
+
+	/**
+	 * Plugin's author
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	public $author;
+
+	/**
+	 * Plugin's Author's URI
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	public $author_uri;
+
+	/**
+	 * Plugin's description
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	public $description;
+
+	/**
+	 * Plugin's title
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var string
+	 */
+	public $title;
+
+	/**
+	 * Plugin's status, will either be active or inactive
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var bool
+	 */
+	public $active;
+
+	/**
+	 * If the Plugin is uninstallable
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var bool
+	 */
+	public $uninstallable;
+
+	/**
+	 * Plugin's auto-update toggle
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var bool
+	 */
+	public $auto_updates_enabled;
+
+	/**
+	 * Plugin Updates, if any
+	 *
+	 * @since 1.1.1
+	 *
+	 * @var array
+	 */
+	public $update;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 1.1.1
+	 *
+	 * @param string $slug           The plugin's slug
+	 * @param array  $plugin_updates The plugin updates site transient
+	 * @param array  $plugin_details The details for the plugin
+	 * @param array  $auto_updates   The auto updates option
+	 */
+	public function __construct( $slug, $plugin_updates, $plugin_details, $auto_updates ) {
```

Comment:
This class is essentially used as an API Resource, but it also could be a useful data object representing an installed plugin. I think it would make for a more useful/clean API if all the parameters other than `$slug` were optional. The rest of the information can be computed inside the class as long as the slug is provided.

Reply:
Made the params optional..